modules/cloudsql-instance/variables.tf

```diff
@@ -245,6 +239,22 @@ variable "root_password" {
   default     = null
 }
 
+variable "ssl" {
+  description = "Setting to enable SSL, set config and certificates."
+  type = object({
+    require_ssl = optional(bool, false)
```

Comment:
I would leave default to `null` as the provider then implements its own default (or the API does).